### ISSUE TITLE
[Reading] Reinforcement Learning Teachers of Test Time Scaling

### ISSUE THREAD

**Issue by hiroaki-1-1-0** [hiroaki-1-1-0/Academic-Paper-Reviews#2]
## 0. Info
- Author: Edoardo Cetin, Tianyu Zhao, Yujin Tang
- Paper: [link](https://www.arxiv.org/abs/2506.08388)

## 1. In a Nutshell
- Instead of using Reinforcement Learning (RL) to train a model to find a solution, it trains a "Reinforcement-Learned Teacher (RLT)" to explain a solution that is provided as training data for Supervised Fine-Tuning of student models.

<img width="1001" height="445" alt="Image" src="https://github.com/user-attachments/assets/f8dd8b66-f373-4d18-bd05-19b45e94b1ec" />

## 2. Novelty / Utility
- This RLT method allows a small teacher model to generate raw distillation data that is more effective for training student models than the post-processed data from larger models.
- The RLT also provided superior "cold-starts" for traditional RL fine-tuning.

## 3. Core Method
- The RLT is trained using RL with a dense reward function composed of two main parts, calculated from the perspective of a separate "student" model:
    1. Solution Understanding: This term rewards explanations that increase the student model's likelihood of predicting the correct solution.
    2. Explanation Clarity: This term penalizes explanations that are unnatural or confusing from the student's perspective (i.e., when the student only sees the question).

<img width="477" height="851" alt="Image" src="https://github.com/user-attachments/assets/064f3cfc-0477-4069-b062-b2e382d5921f" />

## 4. Validation
- Both 7B and 32B students trained on the 7B RLT's raw data consistently outperform students trained on data from larger, post-processed models.

<img width="470" height="393" alt="Image" src="https://github.com/user-attachments/assets/60f85a2c-3d21-4430-8055-63fb5046806e" />

## 5. Limitations / Discussions
- The framework requires access to ground-truth solutions for the training problems.
- The training process involves an additional student model to compute the dense rewards, which adds a layer of complexity to the training setup.
- The paper did not explore scaling the RLT model itself beyond 7B parameters.
- The paper did not explore jointly training the teacher and student models.

## 6. Further Reading
- LLMs Can Easily Learn to Reason from Demonstrations Structure, not content, is what matters!

## 7. Personal Comments
- In this paper, they used an instruction model as a teacher. However, when only a base model can be used for training, is there a way to effectively train it by assigning multiple roles to it?
